File: Backend/TeslaACDC.Data/Models/AudioLibro.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace TeslaACDC.Data.Models;

public class Audiolibro : BaseEntity<int>
{
    public string Titulo { get; set; } = string.Empty;
    public string Duracion { get; set; }
    public string TamañoMB { get; set; }
    public string Portada { get; set; } = string.Empty;
    public string PathArchivo { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public string Narrador { get; set; }
    public string Idioma { get; set; }

    public ICollection<AudiolibroGenero> AudiolibroGeneros { get; set; } = new List<AudiolibroGenero>();
    public ICollection<AudiolibroAutor> AudiolibroAutores { get; set; } = new List<AudiolibroAutor>();

}

File: Frontend/app/models/AudioLibro.ts
export interface AudioLibroResumen {
  id: number;
  titulo: string;
  autor: string;
  genero: string;
  portada: string;
}

export interface AudioLibro extends AudioLibroResumen {
  duracion: string;
  tamañoMB: number;
  narrador: string;
  idioma: string;
  pathArchivo: string;
}

export interface AudioLibroResponse {
  message: string;
  statusCode: number;
  totalElements: number;
  responseElements: AudioLibro[];
}

export interface AudiolibroCrear {
  titulo: string;
  autorId: string;
  generoId: string;
  duracion: string;
  portada: string;
  tamañoMB: string;
  narrador: string;
  idioma: string;
  pathArchivo: string;
}
